sdks/java/core/src/main/java/org/apache/beam/sdk/io/Sink.java

```diff
@@ -241,6 +248,12 @@ public abstract void finalize(Iterable<WriteT> writerResults, PipelineOptions op
      */
     public abstract void open(String uId) throws Exception;
 
+    public abstract void setWindowAndPane(BoundedWindow window, PaneInfo paneInfo) throws Exception;
```

Comment:
I'm not arguing against it or anything, but these setter methods seem incongruous relative to the rest of the API here. Are writers being reused in a novel way? (From what I can tell this just sets up the APIs that runners will later implement). Two other natural places to pass this info would be via the `FileBasedWriteOperation` and the call to `open`. I'm trying to grok the choice.